addons/events/fnc_addPlayerEventHandler.sqf

```diff
@@ -0,0 +1,145 @@
+/* ----------------------------------------------------------------------------
+Function: CBA_fnc_addPlayerEventHandler
+
+Description:
+    Adds a player event handler.
+
+    Possible events:
+        "unit"       - player controlled unit changed
+        "weapon"     - currently selected weapon change
+        "loadout"    - players loadout changed
+        "vehicle"    - players current vehicle changed
+        "turret"     - position in vehicle changed
+        "visionMode" - player changed to normal/night/thermal view
+        "cameraView" - camera mode changed ("Internal", "External", "Gunner" etc.)
+        "visibleMap" - opened or closed map
+
+Parameters:
+    _type      - Event handler type. <STRING>
+    _function  - Function to add to event. <CODE>
+
+Returns:
+    _id - The ID of the event handler. <NUMBER>
+
+Examples:
+    (begin example)
+        _id = ["unit", {systemChat str _this}] call CBA_fnc_addPlayerEventHandler;
+    (end)
+
+Author:
+    commy2
+---------------------------------------------------------------------------- */
+#include "script_component.hpp"
+SCRIPT(addPlayerEventHandler);
+
+params [["_type", "", [""]], ["_function", {}, [{}]]];
+
+private _id = switch (toLower _type) do {
+case ("unit"): {
+    [QGVAR(unitEvent), _function] call CBA_fnc_addEventHandler // return id
+};
+case ("weapon"): {
+    [QGVAR(weaponEvent), _function] call CBA_fnc_addEventHandler // return id
+};
+case ("loadout"): {
+    [QGVAR(loadoutEvent), _function] call CBA_fnc_addEventHandler // return id
+};
+case ("vehicle"): {
+    [QGVAR(vehicleEvent), _function] call CBA_fnc_addEventHandler // return id
+};
+case ("turret"): {
+    [QGVAR(turretEvent), _function] call CBA_fnc_addEventHandler // return id
+};
+case ("visionmode"): {
+    [QGVAR(visionModeEvent), _function] call CBA_fnc_addEventHandler // return id
+};
+case ("cameraview"): {
+    [QGVAR(cameraViewEvent), _function] call CBA_fnc_addEventHandler // return id
+};
+case ("visiblemap"): {
+    [QGVAR(visibleMapEvent), _function] call CBA_fnc_addEventHandler // return id
+};
+default {-1};
+};
+
+// add loop for polling if it doesn't exist yet
+if (isNil QGVAR(pollingLoopAdded) && {_id != -1}) then {
+    GVAR(pollingLoopAdded) = true;
+
+    GVAR(oldUnit) = objNull;
+    GVAR(oldWeapon) = "";
+    GVAR(oldLoadout) = [];
+    GVAR(oldLoadoutNoAmmo) = [];
+    GVAR(oldVehicle) = objNull;
+    GVAR(oldTurret) = [];
+    GVAR(oldVisionMode) = -1;
+    GVAR(oldCameraView) = "";
+    GVAR(oldVisibleMap) = false;
+
+    addMissionEventHandler ["EachFrame", {
```

Comment:
Keep track of the EachFrame handler id so that the polling function can be removed if/when all player event handlers are removed again.